### DISCUSSION TITLE
Tesla token not renewing automatically - requires manual request each day

### DISCUSSION THREAD

**Discussion by OttoKorrect** [pkuehnel/TeslaSolarCharger#1078] [General]
How do I troubleshoot this?

I am running my docker image in Portainer, so whilst I can stream the log, I am not sure how to have the log save to file, which this probably requires. 
Is there a log path in the container I can access, or map to a host volume?

![image](https://github.com/pkuehnel/TeslaSolarCharger/assets/7450377/448e9e83-1e5e-4d9e-8745-ac83a6627031)

**Comment by pkuehnel** [comment]
Very likely this is due to a bug on Tesla's API. I already was in contact with Tesla regarding this issue and they fixed it for another account. Could you please give me the last 5 chars of your installation ID (bottom of the home page of TSC). Note: **Do not share the whole ID** here.

Edit: regarding the logfiles, you can generate one like described [here](https://github.com/pkuehnel/TeslaSolarCharger?tab=readme-ov-file#generate-logfiles). You might look for one of the following messages:

- `Your token or refresh token is invalid. Very likely you have changed your Tesla password.`
- `You did not select all scopes, so TSC can't send commands to your car.`
- `Vehicle {vin} is not paired with TSC. Add The public key to the vehicle`
- `Staus Code {statusCode} is currently not handled, look into https://developer.tesla.com/docs/fleet-api#response-codes to check status code information`

But do not forget to share the last 5 chars of your installation ID, so I can help you further.

**Reply by OttoKorrect** [reply]
a3ea1

**Reply by pkuehnel** [reply]
It's not the issue i expected. Will look deepter into it in about 10 hours. For you to know: The issue should occur always 8 hours after logging in as the token refresh does not work and the lifetime is 8 hours.

**Reply by pkuehnel** [reply]
@pflodo Please recreate the token again. If you still have the issue after 8 hours, please give me your Tesla Account email address, and I will contact Tesla.

Edit: one moment, Looks like I messed up something..

**Reply by pkuehnel** [reply]
@pflodo OK, now you can regenerate the token; sorry for the confusion.

**Reply by OttoKorrect** [reply]
Not sure if something changed, but it is still going fine ~24 hours later.

**Reply by pkuehnel** [reply]
Very likely Tesla fixed the issue for all accounts now.
